test/packaging/ansible/roles/agent-upgrade/tasks/upgrade-RedHat.yaml

```diff
@@ -8,5 +8,16 @@
     state: "latest"
   retries: 5
   delay: 60
+  when: not "python3" in ansible_python_interpreter
```

Comment:
why check python3 and not check when dnf is present? is it a 1to1 rule? 🤔

Reply:
I assume that `dnf` is installed and used whenever python3 is available because:

 [Yum ansible module:](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_module.html) `This module only works on Python 2. If you require Python 3 support see the ansible.builtin.dnf module.`
 
 AL2022 only contains Python 3 and the yum build module failed.

Reply:
👍  good to know